### PR TITLE
Add instructions for building within CESM, linking to FTorch, and running.

### LINKED ISSUE #1: CAM Information
CAM code is at [github.com/ESCOMP/CAM](https://github.com/ESCOMP/CAM).

We want version 6.3.139 which is "mid-top" CAM.:
```
git clone https://github.com/ESCOMP/CAM  CAM
cd CAM
git checkout cam6_3_139
./manage_externals/checkout_externals
```
Qiang is re-training NN to work with this model.

Benchmark to use is FMTHIST.

From Joan:
Context here is that this is more recent "mid-top" CAM capable of modelling stratosphere comfortably.
It is suspected that this is where we will be going in future, so makes sense to work on this.

If you want to run CESM on NCAR, this is one command you will need.
```
./create_newcase --compset FMTHIST --res ne30pg3_ne30pg3_mg17 --case ../../cases/MMdiags --project XXXXXXX --machine derecho
```

- [x] Jack and Surbhi to try and run benchmark for unaltered CAM on Derecho (limit number of timesteps) to check it is set up OK. - Ask Qiang or Martina for help if issues.

### LINKED ISSUE #6: Documentation for building and running
As we start of modify the source code we should provide clear instructions for ourselves, collaborators, and others on how to use this code.

This will require:

- [ ] Obtaining and setting up CESM
- [ ] Using this version of CAM from within CESM
- [ ] Linking to FTorch from within CESM (Using ICCS CIME fork)
- [ ] Building and running a case, setting any options as required in namelists etc.

### PR DESCRIPTION
**Pull request by jatkinson1000**
Closes #6 
Relates to #1, #2 

I have added instructions for
- [x] obtaining CESM and checking out the relevant tag (2.1.5 tbc with Joan/Qiang)
- [x] pointing to use this fork of CAM as part of CESM build
- [x] pointing to the ICCS fork of CIME as part of the CESM build
    - [x] minimal instructions/notes about building FTorch
- [x] Building and running
    - [x] I assume #4 will be merged so make reference to setting the namelist variables we define there (switch and location of model)

Possible TODO:

- [ ] Specify which variables to add to the fincl files.\
        Though perhaps this will come in a later PR as we modify the code and add these?

### CODE REVIEW

**Review by yqsun91 — APPROVED** [review]

**Comment by yqsun91** [comment]
Yes, this is a very good doc.  The only thing I had was the branch for FTorch-compatible CIME.
The doc shows `branch = ftorch_gw`,  If I click the ICCS link, it changes to `branch = ftorch_forpy_cime`.   I am using the ftorch_gw one.

**Comment by jatkinson1000** [comment]
Thanks @yqsun91 
That's useful to hear - the `ftorch_forpy_cime` is the main one used with CAM6 originally in this and other projects.
The newer branch that we use here is the one I had to make to work after there were some updates to CIME as is used in CAM7.

I'll bear this in mind, but am glad you were able to follow the setup without it being a sticking point.

File: README.md [hunk 1]
```diff
@@ -6,10 +6,203 @@ developed as part of the DataWave project.
...
+Before we can run `./case.build` we first need to make some manual changes to the `Makefile` located in
+`<path_to_testcase_directory>/Tools/Makefile`. This will allow the CIME build system to locate `FTorch`.
+
+From the test case directory, modify `Tools/Makefile` line 602 to set the environment variable `FTORCH_LIB` to the location of
+the `FTorch` library on your system.
+
+If you followed the instructions in section [`FTorch` on Derecho](#ftorch-on-derecho) this will be `$HOME/FTorch/bin/ftorch_intel`.
+
+```make
+FTORCH_LIB := $HOME/FTorch/bin/ftorch_intel
```
Comment (j-emberton) [inline README.md]: Do we want to alter this to the full glade path instead of $HOME as was required to get my build working properly?

**Review by TomMelt — APPROVED** [review]
Thanks @jatkinson1000 . LGTM :+1:

File: README.md (hunk 1 above)
Comment (jatkinson1000) [reply README.md]: Does https://github.com/DataWaveProject/CAM/pull/5/commits/5cfb8f925efb50eaa91dd907df7436a47ab2f74d resolve this?

**Review by j-emberton — APPROVED** [review]
Looks good

**Pull request merged**